runtime/compiler/optimizer/StringPeepholes.cpp

```diff
@@ -79,6 +79,53 @@
 #include "env/PersistentCHTable.hpp"           // for CLASSHASHTABLE_SIZE, etc
 #include "env/PersistentInfo.hpp"              // for PersistentInfo
 
+void TR_StringPeepholes::removePotentialOSRPointHelperCalls(TR::TreeTop* start, TR::TreeTop* end)
+   {
+   if (comp()->isOSRTransitionTarget(TR::postExecutionOSR) && comp()->getOption(TR_EnableOSR))
+      {
+      TR::TreeTop *tt = start;
+      do
+         {
+         TR::Node *osrNode = NULL;
+         if (comp()->isPotentialOSRPoint(tt->getNode(), &osrNode))
+            {
+            if (osrNode->isPotentialOSRPointHelperCall())
+               {
+               if (trace())
+                  traceMsg(comp(), "Remove tt n%dn with potential osr point %p n%dn\n", tt->getNode()->getGlobalIndex(), osrNode, osrNode->getGlobalIndex());
+               TR::TreeTop* prevTT = tt->getPrevTreeTop();
+               TR::TransformUtil::removeTree(comp(), tt);
+               tt = prevTT;
+               }
+            }
+         tt = tt->getNextTreeTop();
+         } while (tt != end);
+      }
+   }
+
+void TR_StringPeepholes::fixUpOSRBookkeeping(TR::TreeTop* start, TR::TreeTop* end)
```

Comment:
This is a bit misleading I think this needs to be something like prohibitOSROverRange or something like that since the bookkeeping is the stuff in the OMR layer tracking the live slots etc.